Rename keys into authentication (#236)
meilisearch/documentation: guides/advanced_guides/authentication.md

# Authentication

MeiliSearch uses a key-based authentication. There are three types of keys:

- The **Master** key grants access to all routes.
- The **Private** key grants access to all routes except the `/keys` routes.

> - The **Public** key grants only access to the following routes:

Reviewer: ```suggestion
- The **Public** key only grants access to the following routes:
```

  - `GET /indexes/:index_uid/search`
  - `GET /indexes/:index_uid/documents`
  - `GET /indexes/:index_uid/documents/:doc_id`

When a master key is provided to MeiliSearch, both the private and the public keys are automatically generated. **You cannot create any additional keys**.

### Master Key

When launching an instance, you have the option of giving a master key. By doing so, all routes will be protected and will require a key to be accessed.

You can specify it by passing the `MEILI_MASTER_KEY` environment variable, or using the command line argument `--master-key`.

> Thanks to the master key, you can retrieve both the private and the public keys using the [keys route](/references/keys.md).

Reviewer: Is thanks not used in an reassuring kind of way?
```suggestion
You can retrieve both the private and the public keys using the master key on the [keys route](/references/keys.md).
```

Author: Much better, thanks!

> On each API call, the master key must be added to [the header](/references/#authentication).

Reviewer: That's a different sentence. Since you have 3 keys, you can put either one of the keys in the header. 
If you put the public key, you will be able to use the /search route
If you put the private key you will be able to use all the routes expect the key route
and if you use the master key you will be able to use all routes.
```suggestion
## API Key

If a master key is set, on each API call, a key must be added to [the header](/references/#authentication).
If no or a wrong API key is provided in the header you will have no access to any route and you will receive the
`HTTP/1.1 403 Forbidden` status code.

```

If you like this suggestion you could put it under #### No master key

Author: Yes, sure! Do I add `## API Key` somewhere?

Reviewer: Maybe just before ## Reset Keys ?
Oh and by the way, could you change `### Master Key` and `### Reset Keys` to `## Master Key` and  `## Reset Key`

Author: Done!
And removed the former line:

```
On each API call, the master key must be added to [the header](/references/#authentication).
```

#### No master key

If no master key is provided, all routes can be accessed without requiring any key.

### Reset keys

Since both the private and the public keys are generated based on your master key, changing the master key will result in the modification of the two other keys.

> After having changing reset master key, it is mandatory to restart the MeiliSearch server to ensure the renewal of the private and the public keys.

Reviewer: Not sure what you wanted to say here.
```suggestion
After having changed your master key, it is mandatory to restart the MeiliSearch server to ensure the renewal of the private and the public keys.
```
